### DISCUSSION TITLE
Problem with sort()

### DISCUSSION THREAD

**Discussion by therealchrismcfly** [neuefische/questions#86] [Web] [answered]
Hello, 

I have a problem with sorting an Array of Objects.  

**My goal:**
- When the user clicks on the checkbox of an exercise card on the planner page, the checked exercise automatically moves to the bottom of the list. 
<img width="396" alt="Bildschirmfoto 2022-08-19 um 12 25 38" src="https://user-images.githubusercontent.com/107792731/185599361-4d751fff-baa7-4722-a73b-1a46f9b5d9b0.png">

**Here's what happened:**
- I added a Checkbox to the cards on the planner page (components/Cardlist/planner.js). 
- I used the sort function like this: 
![Bildschirmfoto 2022-08-19 um 12 21 48](https://user-images.githubusercontent.com/107792731/185598851-f7a1e96f-b2c3-4e94-a4e1-7bf682c00ab1.png)

Here is the link to my GitHub Repo: 
https://github.com/therealchrismcfly/capstone-project/tree/addcheckbox

Thanks a lot in advance!
Chris

**Comment by doemser** [comment, marked as answer]
Hey Chris,
take a look at this prototype, maybe this will help you :)

https://codesandbox.io/s/sort-array-on-check-x593g4?file=/src/App.js

**Reply by therealchrismcfly** [reply]
ich hab es genau so gemacht wie in der Codesandbox und es klappt nicht. :(

**Reply by doemser** [reply]
Wir haben den Fehler gefunden:

Deine Checkbox Component müsste so aussehen, damit es funktioniert:

```js
export default function Checkbox({checked, onChange}) {
    return <input type="checkbox" checked={checked} onChange={onChange} />;
}
````
AAABER tu das nicht, ersetze die `<Checkbox/>` lieber durch eine StyledCheckbox

**Reply by therealchrismcfly** [reply]
Ok, jetzt klappt es bei meine Initialen Daten, aber nicht bei den exercise Karten, die neu hinzugefügt werden.

**Reply by doemser** [reply]
Lass uns kurz im zoom dazu treffen.

**Reply by therealchrismcfly** [reply]
danke <3
